Makefile.config

```diff
@@ -62,15 +59,15 @@ endif
 # concurrencykit was installed via Ports, it won't be picked up without explicitly looking
 # for it here
 ifeq ($(OS),FreeBSD)
-LDFLAGS += -L/usr/local/lib
-CFLAGS  += -I/usr/local/include
+COMP_LIBDIRS += -L/usr/local/lib
+COMP_INCDIRS += -I/usr/local/include
 endif
 
 
 # Commands "baked into" cyclone for invoking the C compiler
-CC_PROG ?= "$(CC) ~src-file~ $(COMP_PROG_CFLAGS) -c -o ~exec-file~.o"
-CC_EXEC ?= "$(CC) ~exec-file~.o ~obj-files~ $(LIBS) $(COMP_CFLAGS) -o ~exec-file~"
-CC_LIB  ?= "$(CC) ~src-file~ $(COMP_CFLAGS) -c -o ~exec-file~.o"
+CC_PROG ?= "$(CC) ~src-file~ $(COMP_PROG_CFLAGS) ~cc-extra~ $(COMP_INCDIRS) -c -o ~exec-file~.o"
+CC_EXEC ?= "$(CC) ~exec-file~.o ~obj-files~ $(LIBS) $(COMP_CFLAGS) ~ld-extra~ $(COMP_LIBDIRS) -o ~exec-file~"
+CC_LIB  ?= "$(CC) ~src-file~ $(COMP_CFLAGS) ~cc-extra~ $(COMP_LIBDIRS) -c -o ~exec-file~.o"
```

Comment:
The difference between `CC_LIB` and `CC_PROG` is unclear to me. Shouldn't it be sufficient to have a single "rule" for compiling object files, independent of whether these object files are ultimately linked together as a shared library or an executable?

Reply:
By separating these into separate options we can have specific flags that only make sense for programs and/or libraries. For example `-PIC` is not required for programs: https://github.com/justinethier/cyclone/blob/d2915abe6a39ba008f7c2829f13a3374ae43ef9d/Makefile.config#L33